### ISSUE TITLE
mnist classification raises runtime error

### ISSUE THREAD

**Issue by TomonobuTsujikawa** [sony/nnabla-examples#233]
When running mnist classification, "modifying data is prohibited" error occurs in v1.20.0 and later.

```
nnabla-examples/mnist-collection$ python classification.py --context=cudnn
2021-08-04 09:00:00,652 [nnabla][INFO]: Initializing CPU extension...
2021-08-04 09:00:00,974 [nnabla][INFO]: Running in cudnn
2021-08-04 09:00:00,028 [nnabla][INFO]: Initializing CUDA extension...
2021-08-04 09:00:00,063 [nnabla][INFO]: Initializing cuDNN extension...
Traceback (most recent call last):
  File "classification.py", line 238, in <module>
    train()
  File "classification.py", line 154, in train
    pred = mnist_cnn_prediction(image, test=False, aug=args.augment_train)
  File "classification.py", line 59, in mnist_lenet_prediction
    c1 = F.relu(F.max_pooling(c1, (2, 2)), inplace=True)
  File "<relu>", line 3, in relu
  File "~/envs/nnabla-20/lib/python3.6/site-packages/nnabla/function_bases.py", line 999, in relu
    return F.ReLU(ctx, inplace)(x, n_outputs=n_outputs, auto_forward=get_auto_forward(), outputs=outputs)
  File "function.pyx", line 317, in nnabla.function.Function.__call__
  File "function.pyx", line 295, in nnabla.function.Function._cg_call
RuntimeError: value error in check_data_inplace
nnabla/src/nbla/computation_graph/function.cpp:83
Failed `input->allow_modify_data()`: Modifying data is prohibited by the parent function of the 0-th input data of 'ReLUCudaCudnn' (depth=3). (Parent is 'MaxPoolingCudaCudnn').
```

**Comment by TomonobuTsujikawa** [comment]
Now we prepared PRs to ignore inplace option from ReLU, LeakyReLU, RandomErase, Add2, BcAdd2, Sub2, AddScalar, MulScalar.
Please be careful if you expect inplace behavior.
* https://github.com/sony/nnabla/pull/950
* https://github.com/sony/nnabla-ext-cuda/pull/336

This change will be released by nnabla v1.21.0.

**Comment by TomonobuTsujikawa** [comment]
v1.21.0 was released, so close this issue.
